src/graph/Executor.cpp

```diff
@@ -218,5 +220,43 @@ Status Executor::checkFieldName(std::shared_ptr<const meta::SchemaProviderIf> sc
     return Status::OK();
 }
 
+StatusOr<int64_t> Executor::toTimestamp(const VariantType &value) {
+    if (value.which() != VAR_INT64 && value.which() != VAR_STR) {
+        return Status::Error("Invalid value type");
+    }
+
+    int64_t timestamp;
+    if (value.which() == VAR_STR) {
+        std::string timeStr = boost::get<std::string>(value);
+        static const std::regex reg("^[1-9]\\d{3}-"
+                                    "(0[1-9]|1[0-2]|\\d)-"
+                                    "(0[1-9]|[1-2][0-9]|3[0-1]|\\d)\\s+"
+                                    "(20|21|22|23|[0-1]\\d|\\d):"
+                                    "([0-5]\\d|\\d):"
+                                    "([0-5]\\d|\\d)$");
+        std::smatch result;
+        if (!std::regex_search(timeStr, result, reg)) {
+            return Status::Error("Invalid timestamp type");
+        }
+        struct tm time;
+        memset(&time, 0, sizeof(time));
+        sscanf(timeStr.c_str(), "%d-%d-%d %d:%d:%d",
+               &time.tm_year, &time.tm_mon, &time.tm_mday,
+               &time.tm_hour, &time.tm_min, &time.tm_sec);
+
+        time.tm_year -= 1900;
+        time.tm_mon--;
+        timestamp = mktime(&time);
+    } else {
+        timestamp = boost::get<int64_t>(value);
+    }
+
+    static const int64_t maxTimestamp = std::numeric_limits<int64_t>::max() / 1000000000;
```

Comment:
Please add a comment to explain this threshold, like _The mainstream Linux kernel's implementation constrains this_